pandas/core/internals/blocks.py

```diff
@@ -496,6 +500,10 @@ def astype(
                 f"({self.dtype.name} [{self.shape}]) to different shape "
                 f"({newb.dtype.name} [{newb.shape}])"
             )
+        if using_copy_on_write():
+            if not copy:
```

Comment:
cc @jorisvandenbossche 

I see 2 options here: Creating a function that checks if original and target dtype can be cast to each other without making a copy or figuring out a way to check for shared memory. Thoughts?

Reply:
Checking numpy's astype, I think it always returns a copy _except_ if the dtype is exactly the same (eg casting int64 to datetime64[ns], would could be a view, is still a copy even with `copy=False`). 

So _if_ we don't have special fast paths for such case ourselves, it might be possible to rely on just checking if dtypes are equal (but we would have to check our extension types is to check that)

Reply:
Casting from int64 to Int64 is a view unfortunately, so this has to be more intelligent